### PR TITLE
support gamepad triggers as buttons

### LINKED ISSUE #11192: Nintendo Switch Controller false mapped buttons
**Describe the bug/issue**
I want to use the wired Nintendo Switch Controller made by PowerA.
lsusb = ID 20d6:a711
[https://www.powera.com/p/nintendo/nintendo-switch/controllers/wired/wired-controller-for-nintendo-switch-white-1517033-01/](url)

But some buttons and sticks are mapped wrong and some are not recogized.
controller   >>>   darktable
dpad left = is ok
dpad right = is ok
dpad up = is ok
dpad down = is ok
button B = is ok
button X = is ok
left stick movement = is ok

button A = not recognized
button Y = button A
left shoulder = button Y
right shoulder = not recogized
left trigger = left shoulder
right trigger = right shoulder
left stick press = button back
right stick press = button start
small sqare button = left stick
small home button = button guide
small round - button = not recognized
small round + button = not recognized

* darktable version : e.g. 3.8.1 from OBS build service (latest release)
* OS                : e.g. Linux -  5.11.0-46-generic
* Linux - Distro    : e.g. Kubuntu 20.4
* Desktop           : Kde Plasma

**Additional context**
 - Can you reproduce with another darktable version(s)?
 yes with 2.9 snapshot from the master branch from OBS

 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? 
 yes

 in Windows 10 the same buttons are not or wrong recognized

### PR DESCRIPTION
**Pull request by dterrahe**
fixes #11192

As per https://github.com/darktable-org/darktable/issues/11192#issuecomment-1046898557 some gamepad have on/off triggers instead of "normal" triggers that send variable (but all positive) values. Previously I didn't know what to do with triggers since the one-direction nature makes them pretty useless to control sliders etc. But now I've translated them to "normal" key presses. If variable values are received, then single/double/triple presses are generated depending on how far and fast the trigger is pressed. This may or may not be useful, but if double or triple presses are not mapped, then all presses are interpreted as single presses anyway (as per normal Input NG behavior).

This also avoids people getting confused that only triggers don't work and wasting time trying to "fix" the problem, when it was simply not supposed to work/unsupported.

@every-good-name-is-already-taken would you mind testing if this works for you? Many thanks!

Also, this corrects a small issue where if multiple controllers are connected, the buttons and joysticks didn't use the same id except for the first one.

### CODE REVIEW

**Comment by dterrahe** [comment]
I'll see if I can make this configure using `dt_capabilities_add()` at the same time as #11361 so please hold off merging for now.

**Comment by dterrahe** [comment]
Actually, I misunderstood `dt_capabilities_add` to maybe also link into the cmake configuration. Since it only turns on/off preferences, and since gamepads don't have any settings, it is not needed here.

I did push an extra change here; after adjusting midi parameters or adding devices, the input system can be restarted with ctrl+shift+alt+I. This first closed down SDL2 via `SDL_Quit`, on the assumption it could then restarted again with `SDL_Init` but that doesn't work. So I've commented out that one line and now it works fine for me (on Debian); adding controllers and restarting makes them available. The order is somewhat random, but since I highly doubt many people are using dt as a multi-player game, I'm not too bothered...

File: src/libs/tools/gamepad.c
Comment (TurboGit) [inline src/libs/tools/gamepad.c:321]: Is that some debug left over?

Comment (dterrahe) [reply src/libs/tools/gamepad.c]: > I did push an extra change here; after adjusting midi parameters or adding devices, the input system can be restarted with ctrl+shift+alt+I. This first closed down SDL2 via SDL_Quit, on the assumption it could then restarted again with SDL_Init but that doesn't work. So I've commented out that one line and now it works fine for me (on Debian); adding controllers and restarting makes them available. The order is somewhat random, but since I highly doubt many people are using dt as a multi-player game, I'm not too bothered...

**Review by TurboGit — APPROVED** [review]
Looks good to me, thanks!

**Comment by TurboGit** [comment]
I'll be waiting for @every-good-name-is-already-taken feedback before merging.

**Comment by dterrahe** [comment]
> I'll be waiting for @every-good-name-is-already-taken feedback before merging.

We haven't heard from them in a while and anyway, they might not be able to compile from source/a PR branch, so it might be better to merge and then get confirmation it fixed #11192. This is a localized change to gamepad so not high risk.

**Comment by TurboGit** [comment]
> > I'll be waiting for @every-good-name-is-already-taken feedback before merging.
> 
> We haven't heard from them in a while and anyway, they might not be able to compile from source/a PR branch, so it might be better to merge and then get confirmation it fixed #11192. This is a localized change to gamepad so not high risk.

Ok, let's merge this. Indeed no big risk. Thanks.

**Pull request merged**
